### PR TITLE
[Linux] enable most functional tests on Linux

### PR DESCRIPTION
**Pull request by chrisd8088**
This is an initial set of changes to enable most functional tests to run -- if not succeed -- on Linux.

We build the `GVFS.FunctionalTests` projects on Linux, change a number of `MacTODO` categories into `NonWindowsTODO` equivalents, and deal with some Linux-specific failures caused by the present lack of a `GVFS.Service` daemon and the temporary use of `flock(2)` in libprojfs (see https://github.com/github/libprojfs/issues/79 for details).

Enabling additional functional tests on Linux depends on the work in #1411, #1413, and especially #1412.

/cc @kivikakk, @wilbaker

### CODE REVIEW

File: GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/MultithreadedReadWriteTests.cs [hunk 1]
```diff
@@ -133,32 +164,69 @@ public void CanReadWriteAFileInParallel(FileSystemRunner fileSystem)
...
+                    while (retry)
                     {
-                        while (keepRunning)
+                        Stream readStream = null;
+                        try
                         {
-                            Thread.Yield();
-                            fileContents[myIndex].Append(reader.ReadToEnd());
+                            readStream = File.Open(virtualPath, FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
+                            retry = false;
```
Comment (derrickstolee) [inline GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/MultithreadedReadWriteTests.cs]: Am I reading this right that this loop will repeat infinitely if `File.Open()` throws? What's the benefit there?

Comment (derrickstolee) [reply GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/MultithreadedReadWriteTests.cs]: (and that's the only way it will loop at all)

Comment (chrisd8088) [reply GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/MultithreadedReadWriteTests.cs]: Good catch and question -- the workaround here is to deal with `EAGAIN` from the `File.Open()`.  I think I'm missing some additional handling around the non-`EAGAIN` case, where we want to exit the loop.  Stay tuned and thanks!

**Pull request merged**